### ISSUE TITLE
Foreman external pillar doesn't load

### ISSUE THREAD

**Issue by peripatetic-sojourner** [saltstack/salt#31704]
### Description of Issue/Question

There is a bug in using foreman-node as the external pillar which prevents successful state execution, as the pillar data returns empty. I haven't dug into whether this is an error on the part of Foreman or SaltStack. The issue for that is [here](http://projects.theforeman.org/issues/13558). 

As a work-around I switched us to using the foreman pillar. The foreman.py code would not load, due to **virtualname** and json functionality failures.
### Steps to Reproduce Issue

Follow the documentation to add foreman as the external pillar. Try to refresh pillar data.

**Comment by jfindlay** [comment]
@peripatetic-sojourner, thanks for reporting.  What version of salt are you using?

**Comment by peripatetic-sojourner** [comment]
Sorry, for taking so long to get back. 2015.8.7

Work has been hectic.  Anyway, is there a reason the foreman pillar doesn't use the same file that foreman-node utilizes?  Rather than accessing the api (current method) it would be possible to call https://foreman/salt/node/<minion_id>?format=yml which presents all the same information as foreman-node retrieves.

I'm not familiar enough with Ruby or Foreman code base to fix foreman-node, but I did create a couple replacement python scripts for master_tops and ext_pillar, to get everything running the same way using the above url.  They won't upload grains, but it provides a workaround in case foreman-node breaks. If you want me to submit a merge request for that I can, otherwise the merge request I made a couple days would close this issue.
